library/lib-core-logging/src/main/java/com/sokima/lib/corelogging/PatternMaskingLayout.java

```diff
@@ -0,0 +1,41 @@
+package com.sokima.lib.corelogging;
+
+import ch.qos.logback.classic.PatternLayout;
+import ch.qos.logback.classic.spi.ILoggingEvent;
+import com.sokima.lib.corelogging.usecase.MaskUseCase;
+import com.sokima.lib.corelogging.usecase.impl.JsonMaskUseCaseImpl;
+import com.sokima.lib.corelogging.usecase.impl.ToStringMaskUseCaseImpl;
+
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class PatternMaskingLayout extends PatternLayout {
+    private static final String DELIMETER = ",";
+    private MaskUseCase jsonMaskUseCase;
+    private MaskUseCase toStringMaskUseCase;
```

Comment:
SOLID - Open Closable Principle Violation 
When we got a new type of messages (for example XML), then we need to change PatternMaskingLayout. 

```suggestion
    private MaskUseCase jsonMaskUseCase;
    private MaskUseCase toStringMaskUseCase;
```

Reply:
consider using Strategy Pattern, or Chain Of Responsibility